### PR TITLE
fix: check if installed version is more recent than version specified in version stream

### LINKED ISSUE #7274: Confused versions
### Summary 

I'm trying to upgrade my cluster with `jx boot upgrade`.

Early on, it asks me the following:

```
WARNING: jx version specified in the version stream /Users/vfarcic/.jx/jenkins-x-versions is 2.0.1202. You are using 2.1.56. We highly recommend you upgrade to it.
```

Should I upgrade from `2.1.56` to `2.0.1202`?

### PR DESCRIPTION
**Pull request by ankitm123**
Signed-off-by: ankitm123 <ankitmohapatra123@gmail.com>

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Special notes for the reviewer(s)
Modified the verifyJXVersion function to accept interfaces instead of structs. Accepting interfaces and returning structs will make testing simpler.

#### Which issue this PR fixes

fixes #7274

### CODE REVIEW

File: pkg/cmd/opts/install.go [hunk 1]
```diff
@@ -382,7 +382,8 @@ func (o *CommonOptions) GetLatestJXVersion(resolver *versionstream.VersionResolv
 		return util.GetLatestVersionStringFromBucketURLs(latestVersionStrings)
 	}
 
-	dir := resolver.VersionsDir
+	dir := resolver.GetVersionsDir()
+	fmt.Println("Gone past this")
```
Comment (abayer) [inline pkg/cmd/opts/install.go:386]: Pretty sure you meant to remove this. =)

Comment (ankitm123) [reply pkg/cmd/opts/install.go:386]: LOL :+1:

File: pkg/versionstream/interface.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package versionstream
+
+// VersionStreamer contains the functions related to version stream
```
[the remark this answers is not in the document]
Comment (abayer) [reply pkg/versionstream/interface.go:3]: I _think_ this should be the last line before the type definition with the `go:generate` above that.

File: pkg/cmd/opts/install.go (hunk 1 above)
Comment (ankitm123) [reply pkg/cmd/opts/install.go]: Did a git commit and push from a different directory (pkg/cmd ..) instead of the base jx directory

**Comment by abayer** [comment]
@ankitm123 Looks like you need to do a `gofmt`. =)

File: pkg/versionstream/interface.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ankitm123) [reply pkg/versionstream/interface.go]: I see in majority places in the codebase, `go:generate` comes below the comment ... should we change that?

[the remark this answers is not in the document]
Comment (ankitm123) [reply pkg/versionstream/interface.go]: So the issue was I changed the name from versionStreamer to Streamer, but never changed it in the comments.

**Pull request merged**